### PR TITLE
Scroller widget

### PR DESCRIPTION
**Pull request by emilybram**
PR is bigger than it looks, lots of the lines changed are from Prettier formatting existing code :) 

---

open-oakland-scroll

### CODE REVIEW

**Comment by vahurtad** [comment]
This looks good @emilybram 🎉.
Although, the last scrolling dot does not fill when my screen is bigger. (it works properly when I resize it smaller)
Could this be another issue?

scroll-widget1

File: src/components/central-program.js [hunk 1]
```diff
@@ -41,73 +53,77 @@ const CentralProgram= ({ data }) => {
...
-        </Layout>
-    )
+        <Container>
+          <Row>
+            <Col md={9} xl={6} className="mx-auto">
+              <div id={`${ELEMENT_NAME_PREFIX}-1`} className="pt-4 pt-sm-4">
+                <h2 className="pb-3 pt-sm-3 pt-md-2">{`Staff Roles (${data.site.siteMetadata.latestSchoolYear})`}</h2>
+                <StaffRolesTable data={centralProgram.staff_roles} />
+              </div>
+              <div id={`${ELEMENT_NAME_PREFIX}-2`} className="pt-4">
```
Comment (jbaldo) [inline src/components/central-program.js:63]: so if we add a new section, we just give it an id like this and up the count here and the total in `numSections`?

File: src/components/central-program.js [hunk 2]
```diff
@@ -41,73 +53,77 @@ const CentralProgram= ({ data }) => {
...
+      code
+      staff_roles {
+        eoy_total_positions_for_role
+        role_description
+      }
+      staff_bargaining_units {
+        eoy_total_positions_for_bu
+        description
+        abbreviation
+      }
```
Comment (jbaldo) [inline src/components/central-program.js:104]: Sorry, the code formatting is my fault. I've had my tab width set to 4. Didn't realize we had a Prettier config. It must have come with the Gatsby starter project. I'll try to use that from now on.

File: src/components/central-program/central-program.scss [hunk 3]
```diff
@@ -1,26 +1,31 @@
...
+      div.value {
+        font-weight: normal;
+      }
     }
+  }
+}
+
+.scroll-widget {
+  position: fixed;
+  right: 5%;
```
Comment (jbaldo) [inline src/components/central-program/central-program.scss:30]: Is it possible to get the cursor to turn into a pointer on the dot things? I guess this means it's not an accessible link. :/ But maybe there's not much we can do about that given the design of this widget.

File: src/components/central-program.js (hunk 1 above)
Comment (emilybram) [reply src/components/central-program.js:63]: Yep! It's a little annoying overhead to make sure they are consistent, but I couldn't think of a better way to allow `ScrollWidget` to be a separate component, and it felt a little heavy to define it within `central-program.js`.

File: src/components/central-program.js (hunk 2 above)
Comment (emilybram) [reply src/components/central-program.js:104]: No problem!

File: src/components/central-program/central-program.scss (hunk 3 above)
Comment (emilybram) [reply src/components/central-program/central-program.scss:30]: Yep! Just added another commit to make the cursor a pointer 👍

**Comment by emilybram** [comment]
@vahurtad thanks for catching that! I'm also able to replicate. I've looked into it a bit and the dots are "active" (colored blue) when the corresponding div is at the very top of the screen. This means that:
1. There are super small segments of the screen _right between_ the navigable elements that are not caught by any of the dots, so none of them will be filled in in those cases
2. If the screen is big enough (or the last navigable element is small enough) such that even if you scroll the bottom of the page, the last navigable element is not at the very top, then the bottom dot (which corresponds to the last navigable element) will never be filled in

I think the behavior you found is a combination of both of these -- when the page scrolls to the very bottom, it's landing on a tiny amount of space between the middle & bottom navigable elements.

I'm going to try fixing (1) by moving the navigable element IDs to a higher level, so there is no space between the different elements. I don't think (2) is easily fixable -- would it feel like OK behavior if the middle dot was filled in when you tapped on the bottom dot with a really big screen?

**Pull request merged**